nexus/db-queries/src/db/queries/region_allocation.rs

```diff
@@ -116,24 +122,42 @@ pub fn allocation_query(
     SELECT
       dataset.pool_id,
       sum(dataset.size_used) AS size_used
-    FROM dataset WHERE ((dataset.size_used IS NOT NULL) AND (dataset.time_deleted IS NULL)) GROUP BY dataset.pool_id),")
-
-    // Any zpool already have this volume's existing regions?
-    .sql("
-  existing_zpools AS (
-    SELECT
-      dataset.pool_id
-    FROM
-      dataset INNER JOIN old_regions ON (old_regions.dataset_id = dataset.id)
-  ),")
+    FROM dataset WHERE ((dataset.size_used IS NOT NULL) AND (dataset.time_deleted IS NULL)) GROUP BY dataset.pool_id),");
+
+    let builder = if let Some(snapshot_id) = snapshot_id {
```

Comment:
(No action needed) nice to see, based on tests/output, that this doesn't change the SQL for the existing snapshot-less queries

Reply:
:100: